cmd/fetchd/cmd/stakereconciliationmigrate.go

```diff
@@ -0,0 +1,320 @@
+						oldAccount = _account
+						if newAccount == nil {
+							continue
+						}
+						break
+					}
+
+					if _account.GetAddress().Equals(newFetchAddr) {
+						newAccount = _account
+						if oldAccount == nil {
+							continue
+						}
+						break
+					}
+
+				}
+
+				if oldAccount == nil {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("unable to find old account with address %q", oldFetchAddr.String()),
+					})
+					continue
+				}
+
+				if newAccount == nil {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("unable to find new account with address %q", newFetchAddr.String()),
+					})
+					continue
+				}
+
+				if newAcctBalance == nil {
+					errs = append(errs, fmt.Errorf("new account with address %q does not have a balance", newFetchAddr.String()))
+					continue
+				}
+
+				if oldAccount.GetSequence() != 0 {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     errNonZeroSeqNum,
+					})
+					continue
+				}
+
+				migrateCoin, err := sdk.ParseCoinNormalized(fmt.Sprintf("%s%s", row[3], coinDenom))
+				if err != nil {
+					errs = append(errs, fmt.Errorf("unable to parse amount of tokens to migrate for row %q: %w", row, err))
+					continue
+				}
+				migrateCoins := sdk.NewCoins(migrateCoin)
+
+				if !oldAcctBalance.IsEqual(migrateCoins) {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     errBalanceNotMatch,
+					})
+					continue
+				}
+
+				// Zero out old account balance
+				bankGenesis.Balances[oldBalanceIndex].Coins = sdk.NewCoins()
+
+				// Add migrated coins to new account balance
+				newAcctBalance = newAcctBalance.Add(migrateCoins...)
+				bankGenesis.Balances[newBalanceIndex].Coins = newAcctBalance
+
+				// Mark this ethereum address as migrated.
+				migratedEthAddrs[ethAddrString] = struct{}{}
+
+				// TODO:
+				// - check supply invariant
+			}
+
+			updatedBankGenesisJSON, err := cdc.MarshalJSON(bankGenesis)
+			if err != nil {
+				return fmt.Errorf("unable to marshal updated bank genesis state: %w", err)
+			}
+
+			authGenesis.Accounts, err = authtypes.PackAccounts(genesisAccounts)
+			if err != nil {
+				return fmt.Errorf("unable to pack accounts: %w", err)
+			}
+			updatedAuthGenesisJSON, err := cdc.MarshalJSON(&authGenesis)
+			if err != nil {
+				return fmt.Errorf("unable to marshal updated auth genesis state: %w", err)
+			}
+
+			appState[banktypes.ModuleName] = updatedBankGenesisJSON
+			appState[authtypes.ModuleName] = updatedAuthGenesisJSON
+
+			// build and print the new genesis state
+			genDoc.AppState, err = json.Marshal(appState)
+			if err != nil {
+				return errors.Wrap(err, "failed to JSON marshal migrated genesis state")
+			}
```

Comment:
I'm noticing that the capricorn migrate subcommand makes a call to `app.ModuleBasics.ValidateGenesis(...)`. Perhaps that's worth doing here as well?

Reply:
yup good point, we're not messing too much with genesis structure so low chance it got invalid on the way, but doesn't hurt to double check!